### PR TITLE
[docs] Getting Started with TVM: AutoTVM and Matrix Multiply

### PR DESCRIPTION
**Pull request by hogepodge**
This patch moves the matrix multiplcation example tuning with AutoTVM to the tutorial directory, and expands on the content. This follows and builds on the section on TE.

### CODE REVIEW

**Comment by hogepodge** [comment]
@tqchen @merrymercy

File: tutorials/get_started/autotvm_matmul.py [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+# --------------------
+# To use autotvm package in TVM, we need to install some extra dependencies.
+#
+# .. code-block:: bash
+#
+#   pip3 install --user psutil xgboost cloudpickle
+#
+# To make TVM run faster in tuning, it is recommended to use cython as FFI of
+# TVM. In the root directory of TVM, execute (change "3" to "2" if you use
+# python2):
```
Comment (Hzfengsy) [inline tutorials/get_started/autotvm_matmul.py:58]:
```suggestion
# TVM. In the root directory of TVM, execute:
```
TVM no longer supports Python2

File: tutorials/get_started/autotvm_matmul.py [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+# the module is called `autotvm`
+from tvm import autotvm
+
+################################################################################
+# Basic Matrix Multiplication with TE
+# -----------------------------------
+# Recall the basic implementation of matrix multiplication using TE. We write
+# it down here with a few changes. We will wrap the multiplication in a python
+# function definition.  For simplicity, we will focus our attention on a split
```
Comment (Hzfengsy) [inline tutorials/get_started/autotvm_matmul.py:83]:
```suggestion
# function definition. For simplicity, we will focus our attention on a split
```

File: tutorials/get_started/autotvm_matmul.py [hunk 3]
```diff
@@ -0,0 +1,377 @@
...
+
+# check correctness
+a_np = np.random.uniform(size=(N, L)).astype(np.float32)
+b_np = np.random.uniform(size=(L, M)).astype(np.float32)
+c_np = a_np.dot(b_np)
+
+c_tvm = tvm.nd.empty(c_np.shape)
+func(tvm.nd.array(a_np), tvm.nd.array(b_np), c_tvm)
+
+tvm.testing.assert_allclose(c_np, c_tvm.asnumpy(), rtol=1e-2)
```
Comment (Hzfengsy) [inline tutorials/get_started/autotvm_matmul.py:365]: Why `rtol` is so large here? We usually use `1e-4` ~ `1e-6` for float32

**Review by Hzfengsy — CHANGES_REQUESTED** [review]
Some quick comments

File: tutorials/get_started/autotvm_matmul.py (hunk 3 above)
Comment (hogepodge) [reply tutorials/get_started/autotvm_matmul.py:365]: I don't have an answer for this. We can change to another value that's more consistent with standard practice.

**Comment by hogepodge** [comment]
@tqchen ready for review

**Review by jroesch — APPROVED** [review]

**Pull request merged**